software/controller/lib/hal/hal_stm32.cpp

```diff
@@ -345,8 +346,13 @@ void BadISR() {}
 // the I2C::Channel::*ISR() which are generic ISR associated with an I²C channel
 void I2c1EventISR() { i2c1.I2CEventHandler(); };
 void I2c1ErrorISR() { i2c1.I2CErrorHandler(); };
-void DMA2Channel6ISR() { i2c1.DMAIntHandler(DMA::Channel::Chan6); };
-void DMA2Channel7ISR() { i2c1.DMAIntHandler(DMA::Channel::Chan7); };
+void DMA2Channel6ISR() { i2c1.DMAIntHandler(I2C::ExchangeDirection::Read); };
```

Comment:
took me a little while to figure out why passing the channel is no longer necessary, but I guess the mappings were already there in the i2c implementation, it's just that this part of it was not yet simplified?

Reply:
Mostly, since the I2C itself doesn't remember which DMA::Channel it uses and can't manipulate registers itself anymore, I can't do it the old way.